code/components/citizen-server-impl/src/state/ServerGameState.cpp

```diff
@@ -7074,6 +7074,28 @@ std::function<bool()> fx::ServerGameState::GetRequestControlEventHandler(const f
 #endif
 }
 
+std::function<bool()> fx::ServerGameState::GetScriptEntityStateChangeEventHandler(const fx::ClientSharedPtr& client, net::Buffer&& buffer)
+{
+	uint16_t entityId = 0;
+	uint16_t stateType = 0;
+	rl::MessageBuffer msg;
+
+	if (ParseEvent(std::move(buffer), &msg))
+	{
+		entityId = msg.Read<uint16_t>(13);
+		stateType = msg.Read<uint16_t>(4);
+	}
+
+	return [this, client, entityId, stateType]()
```

Comment:
Are lambda captures needed for this at all? Couldn't you just return a lambda always returning false when the `stateType` is 9 and otherwise return a lambda that is always returning true.

Reply:
Yes that's how I imagined it. Please only add captures if they are absolutely necessary and not for future usage.